src/color.ts

```diff
@@ -0,0 +1,36 @@
+export class Color {
+
+    red: number
+    green: number
+    blue: number
+    alpha: number
+
+    constructor(red: number, green: number, blue: number, alpha: number) {
+      this.red = red ?? 0
+      this.green = green ?? 0
+      this.blue = blue ?? 0
+      this.alpha = alpha ?? 1
+    }
```

Comment:
Since we are using typescript we can utilize the injection from typescript via:
**Note: this is only a suggestion instead of a request, you can decide it on your own whether you want to change it or not.**
```typescript
class Color {
  constructor(public red = 0, public green = 0, public blue = 0, public alpha = 1) {
     // No need to do assignment like this, you can also change public to private if the members were
     // intentionally designed private.
  }

}
```